include/raptor/argument_parsing/validators.hpp

```diff
@@ -83,12 +81,7 @@ class positive_integer_validator
     explicit positive_integer_validator(bool const is_zero_positive_) : is_zero_positive{is_zero_positive_}
     {}
 
-    void operator()(window const & val) const
-    {
-        return operator()(val.v);
-    }
-
-    void operator()(option_value_type const & val) const
+    void operator()(std::integral auto const & val) const
```

Comment:
why not keep
```suggestion
    void operator()(option_value_type const & val) const
```

should be good enough for now

Reply:
Mhm, yes! that should work just fine. (I believed that uint32_t vs size_t would warn, or I misunderstood some error message, also this removes the "Todo")